packages/light-apps/src/Onboarding/Onboarding.tsx

```diff
@@ -8,127 +8,200 @@ import { RouteComponentProps } from 'react-router-dom';
 export class Onboarding extends React.Component<Props, State> {
   state: State = {
-    isBipBusy: true,
+    error: null,
+    mnemonic: '',
     screen: 'importOptions'
   };
 
   componentDidMount () {
-    const seed = generateSeed();
-    const address = generateAddressFromSeed(seed);
+    const mnemonic = generatePhrase();
+    const address = generateAddressFromPhrase(mnemonic);
 
-    this.setState({
-      address: address,
-      isBipBusy: false,
-      seed: seed
-    });
-  }
+    // For Debugging Purposes only
+    if (process.env.NODE_ENV !== 'production') {
+      // @ts-ignore
+      window.keyring = keyring;
+    }
 
-  handleInputSeedPhrase = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) => {
-    this.setState({
-      phrase: value
-    });
-  }
-
-  handleFileUploaded = (file: Uint8Array) => {
-    const json = JSON.parse(u8aToString(file));
-    // FIXME: try catch unlock with json in store
-
-    this.setState({
-      name: json.name,
-      address: json.address,
-      screen: 'save'
-    });
+    this.setState({ address, mnemonic });
   }
 
-  onChangeName = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) => {
+  private onChangeName = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) => {
     this.setState({
       name: value
     });
   }
 
-  onChangePassword = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) => {
+  private onChangePassword = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) => {
     this.setState({
       password: value
     });
   }
 
-  onChangeSeed = ({ target: { value } }: any) => {
+  private onChangePhrase = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) => {
     this.setState({
-      seed: value
+      recoveryPhrase: value
     });
   }
 
-  toggleScreen = () => {
-    const { screen } = this.state;
+  private onError = (value: string | null) => {
     this.setState({
-      screen: screen === 'new' || screen === 'save'
-        ? 'importOptions'
-        : 'new',
-      password: ''
+      error: value
     });
   }
 
-  unlockWithPhrase = () => {
-    // const { phrase } = this.state;
-    // FIXME: try catch unlock with phrase in store
+  private toggleScreen = (to: OnboardingScreenType) => {
     this.setState({
-      screen: 'save',
+      screen: to,
       password: ''
     });
   }
 
-  addAccountToWallet = (file?: Uint8Array) => {
+  private handleFileUploaded = async (file: Uint8Array) => {
+    const {
+      accountStore: { address, name, setIsImport, setJsonString }
+    } = this.props;
+
+    const jsonString = u8aToString(file);
+
+    try {
+      await setIsImport(true);
+      await setJsonString(jsonString);
+
+      this.onError(null);
+
+      this.setState({
+        address,
+        jsonString,
+        name
+      });
+
+      this.toggleScreen('save');
+    } catch (e) {
+      this.onError(e.message);
+      alert('Please resolve this before you continue: ' + e.message);
```

Comment:
Add // FIXME or create a github issue each time there's something that needs to be changed in the future.

Reply:
this one is an alert to the user to resolve an invalid json file upload